pkg/apis/pipeline/v1alpha1/resource_types.go

```diff
@@ -58,8 +61,8 @@ type PipelineResourceInterface interface {
 	Replacements() map[string]string
 	GetDownloadContainerSpec(sourcePath string) ([]corev1.Container, error)
 	GetUploadContainerSpec(sourcePath string) ([]corev1.Container, error)
-	GetUploadVolumeSpec(spec *TaskSpec) ([]corev1.Volume, error)
-	GetDownloadVolumeSpec(spec *TaskSpec) ([]corev1.Volume, error)
+	GetDownloadVolumeSpec(*TaskSpec, *TaskRun, kubernetes.Interface) ([]corev1.Volume, error)
+	GetUploadVolumeSpec(*TaskSpec, *TaskRun, kubernetes.Interface) ([]corev1.Volume, error)
```

Comment:
can we include docstrings for these to explain what the function is expected to do (and what the new params are for?)